rfcs/0006-gtfs-canonical-routes.md

```diff
@@ -0,0 +1,108 @@
+even when stations are closed for a long duration.
+
+PIOs are able to create alerts for stations without any scheduled service, and T-Alerts subscribers
+are able to stay informed about the status of the station closure.
+
+# Guide-level explanation
+
+GTFS output includes additional, hardcoded data about where rail lines go to and where they can stop
+if all stations were open, which is independent of all scheduled service.
+
+This is represented as a service with ID "canonical", which is valid for the entire feed date range,
+but which has no active days of the week. The typicality of the service is tagged with a new value
+so it can be programmatically identified.
+
+All rail routes have two or more trips on this service, corresponding to the "canonical" patterns of
+service in each direction for any branches, as understood by stakeholders. Each such trip has a
+route pattern, for which it is the representative trip, and a shape, and the route pattern is
+considered "typical" for the route. If current service has trips following the same route pattern,
+they will use the canonical route pattern as well.
+
+In short, for consumers of the feed, a route pattern has typicality of `1` if and only if it visits
+a set of canonical stops in a canonical order.
+
+# Reference-level explanation
+
+The canonical service appears in the following GTFS files in the following way:
+
+- **calendar.txt** includes a new `service_id` to envelop all of the canonical trips on all the
+  routes. `start_date` and `end_date` should reflect the feed's valid start/end dates. All other
+  fields should have a value of `0`.
+- **calendar_attributes.txt** includes the new `service_id`. `service_schedule_type` should be
+  `"Other"`, and `rating_start_date` and `rating_end_date` should reflect the feed's valid start/end
+  dates. `service_schedule_typicality` should be `6`, and the
+  [docs](https://github.com/mbta/gtfs-documentation/blob/master/reference/gtfs.md) should be updated
+  with this new kind of value.
+- **route_patterns.txt** includes new route patterns for all routes/directions we want to hardcode.
+  `route_pattern_typicality` should be `1`, and `route_pattern_sort_order` should be a less than the
+  value for all other route patterns for that given route/direction. For routes with branches,
+  you'll need a pattern for each route/direction/branch (eg, for Providence/Stoughton Line, you
+  should have two new route patterns in the outbound direction).
+- **shapes.txt** can probably use the existing shapes.
+- **trips.txt** includes one new trip for each new route pattern. It should reference aforementioned
+  route pattern and shapes. It should reference the new `service_id` created above. Each of these
+  trips should be the `representative_trip_id` for their respective route patterns.
+- **stop_times.txt** includes all stops that trains in the route/direction passes through, even if
```

Comment:
In the same vein as shapes, a long-term subway closure could mean that the closed `stop_id`s could be missing from `build/truncate/stops.txt`, which is why https://github.com/mbta/gtfs_creator/pull/1243 was needed. So we could also have an `input/canonical/stops.txt` that includes the stops...

...But if we go ahead to just manually specify every shape (if we used the same `shape_id`s as revenue trips) and every stop ID in `input/canonical`, we'd now have to drop most of them as duplicates. Would that mean this module is better off coming _right before_ `truncate`, instead of `after`?